transport/handler_server.go

```diff
@@ -40,13 +40,14 @@ import (
 	"google.golang.org/grpc/credentials"
 	"google.golang.org/grpc/metadata"
 	"google.golang.org/grpc/peer"
+	"google.golang.org/grpc/stats"
 	"google.golang.org/grpc/status"
 )
 
 // NewServerHandlerTransport returns a ServerTransport handling gRPC
 // from inside an http.Handler. It requires that the http Server
 // supports HTTP/2.
-func NewServerHandlerTransport(w http.ResponseWriter, r *http.Request) (ServerTransport, error) {
+func NewServerHandlerTransport(w http.ResponseWriter, r *http.Request, stats stats.Handler) (ServerTransport, error) {
```

Comment:
This API change looks good to me, and nobody else should be calling this function.

A slightly different way for this would be to pass in a struct, for example `*transport.ServerConfig`, instead of a `stats.Handler` directly. Also we could unify the constructor of `http2server` and `serverHandlerTransport` into `transport.NewServerTransport`.

But let's leave these for future changes.